### ISSUE TITLE
ZERO DAY EXPLOIT FOUND: Turning off and turning on my monitor, crashes Waylock.

### ISSUE THREAD

**Issue by pragmaticmero** [ifreund/waylock#34]
## Ok hear me out haha.
The lock works as normal, that's until I turn off my monitor or disconnect it and connect it again. Waylock crashes and gives me this error LOL.
```
$ RUST_BACKTRACE=full waylock
[This line is where I turn off and on my monitor]
thread 'main' panicked at 'Cannot send request release which requires version >= 3 on proxy wl_output@9 which is version 1.', /build/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.29.4/src/proxy.rs:85:13
stack backtrace:
   0:     0x563de49a148c - <unknown>
   1:     0x563de49c54fc - <unknown>
   2:     0x563de499e085 - <unknown>
   3:     0x563de49a3010 - <unknown>
   4:     0x563de49a2bc5 - <unknown>
   5:     0x563de49a36c4 - <unknown>
   6:     0x563de49a31a0 - <unknown>
   7:     0x563de49a1934 - <unknown>
   8:     0x563de49a3109 - <unknown>
   9:     0x563de48ada41 - <unknown>
  10:     0x563de497cede - <unknown>
  11:     0x563de4971d37 - <unknown>
  12:     0x563de48bf218 - <unknown>
  13:     0x563de48dead6 - <unknown>
  14:     0x563de48b86e9 - <unknown>
  15:     0x563de48b9381 - <unknown>
  16:     0x563de48bffd0 - <unknown>
  17:     0x563de48d559d - <unknown>
  18:     0x563de48bc941 - <unknown>
  19:     0x563de48e1c73 - <unknown>
  20:     0x563de48d8b4b - <unknown>
  21:     0x563de48e66dc - <unknown>
  22:     0x563de48e3283 - <unknown>
  23:     0x563de48ba859 - <unknown>
  24:     0x563de49a111b - <unknown>
  25:     0x563de48e7922 - <unknown>
  26:     0x7fae4c0b4310 - __libc_start_call_main
  27:     0x7fae4c0b43c1 - __libc_start_main@GLIBC_2.2.5
  28:     0x563de48ae1ee - <unknown>
  29:                0x0 - <unknown>
```
I've run this command with the env "RUST_BACKTRACE=full" before the command, when I run the command without it gives:
```
$ waylock
[This line is where I turn off and on my monitor]
thread 'main' panicked at 'Cannot send request release which requires version >= 3 on proxy wl_output@9 which is version 1.', /build/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.29.4/src/proxy.rs:85:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## System
- Arch Linux 64-Bit
- Installed through pacman
- SwayWM
- wlroots

## Conclusion
This ZERO DAY EXTREMELY DANGEROUS EXPLOIT shouldn't exist, w h y.

**Comment by ifreund** [comment]
The core problem here is with the design of the protocols currently used for screen locking, so I designed a more secure protocol: https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/131.

Once I finish the implementation of that waylock will only support that protocol and bugs such as this will not be a security issue.

**Comment by ifreund** [comment]
Anyhow, fixed in ba336dfb08b70fdb548fa392b3248204fd42b025
